### PR TITLE
Fixed date format in Collect and Submit tables

### PR DESCRIPTION
**Pull request by tienhoah**

### CODE REVIEW

**Comment by tienhoah** [comment]
[sample date of survey - incorrect displayed in submitted summary tables](https://trello.com/c/5wSe3eDi/1245-sample-date-of-survey-incorrect-displayed-in-submitted-summary-tables)

File: src/app/_mermaid_common/forms/directives/dateinput.dir.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ angular
         link: function(scope, element, attr, ngModel) {
           var val;
           var $date_input = element;
-          var dataDateFormat = scope.$parent.$parent.widgetFormat;
+          var dataDateFormat = scope.$parent.widgetFormat;
```
Comment (saanobhaai) [inline src/app/_mermaid_common/forms/directives/dateinput.dir.js]: @tienhoah does this work everywhere, including any modals that might use the directive?

File: src/app/project/controllers/collect-records.ctrl.js [hunk 2]
```diff
@@ -211,11 +211,17 @@ angular.module('app.project').controller('CollectRecordsCtrl', [
           display: 'Sample Date',
           sortable: true,
           formatter: function(v) {
-            let val = '';
-            if (v) {
-              val = $filter('date')(new Date(v), 'dd-MMM-yyyy');
+            let dateResult = '';
+            const dateVal = v && v.split('-').map(val => Number(val));
```
Comment (saanobhaai) [inline src/app/project/controllers/collect-records.ctrl.js]: Can you use `&&` like this? OK if so, but harder to read for me at least

File: src/app/project/controllers/collect-records.ctrl.js [hunk 3]
```diff
@@ -211,11 +211,17 @@ angular.module('app.project').controller('CollectRecordsCtrl', [
...
+            if (dateVal && dateVal.length >= 3) {
+              const newDateVal = new Date(
+                dateVal[0],
+                dateVal[1] - 1,
```
Comment (saanobhaai) [inline src/app/project/controllers/collect-records.ctrl.js]: Why `- 1`? Is the date not correct coming back from the API?

Comment (tienhoah) [reply src/app/project/controllers/collect-records.ctrl.js]: I'm changing to move away from date-string approach to passing number arguments to `new Date()` where 2nd argument is month (which index starts at 0 for January)

File: src/app/_mermaid_common/forms/directives/dateinput.dir.js (hunk 1 above)
Comment (tienhoah) [reply src/app/_mermaid_common/forms/directives/dateinput.dir.js]: scope.$parent.widgetFormat =>  yy-mm-dd 
scope.$parent.$parent.widgetFormat =>  undefined (this happens in Submit record mostly, this is from M1244 bug I found)

File: src/app/project/controllers/collect-records.ctrl.js (hunk 2 above)
Comment (tienhoah) [reply src/app/project/controllers/collect-records.ctrl.js]: Change back to if

**Pull request merged**